usaspending_api/etl/management/commands/load_table_from_delta.py

```diff
@@ -0,0 +1,227 @@
+        else:
+            temp_destination_table_name = f"{delta_table_name}_temp"
+        temp_destination_table = f"{temp_schema}.{temp_destination_table_name}"
+
+        # Resolve JDBC URL for Source Database
+        jdbc_url = get_jdbc_url()
+        if not jdbc_url:
+            raise RuntimeError(f"Couldn't find JDBC url, please properly configure your CONFIG.")
+        if not jdbc_url.startswith("jdbc:postgresql://"):
+            raise ValueError("JDBC URL given is not in postgres JDBC URL format (e.g. jdbc:postgresql://...")
+
+        # Checking if the temp destination table already exists
+        temp_dest_table_exists_sql = f"""
+        SELECT EXISTS (
+                SELECT 1
+                FROM information_schema.tables
+                WHERE table_schema = '{temp_schema}'
+                    AND table_name = '{temp_destination_table_name}')
+        """
+        with db.connection.cursor() as cursor:
+            cursor.execute(temp_dest_table_exists_sql)
+            temp_dest_table_exists = cursor.fetchone()[0]
+
+        # If it does and we're recreating it, drop it first
+        if temp_dest_table_exists:
+            logger.info(f"{temp_destination_table} exists and recreate argument provided. Dropping first.")
+            # If the schema has changed and we need to do a complete reload, just drop the table and rebuild it
+            clear_table_sql = f"DROP TABLE {temp_destination_table}"
+            with db.connection.cursor() as cursor:
+                cursor.execute(clear_table_sql)
+            logger.info(f"{temp_destination_table} dropped.")
+
+        # Recreate the table if it doesn't exist. Spark's df.write automatically does this but doesn't account for
+        # the extra stuff (indexes, constraints, defaults) which CREATE TABLE X LIKE Y accounts for.
+        # If there is no source_table to base it on, it just relies on spark to make it and work with delta table
+        if source_table or source_schema:
+            if source_table:
+                create_temp_sql = f"""
+                    CREATE TABLE {temp_destination_table} (
+                        LIKE {source_table} INCLUDING DEFAULTS INCLUDING IDENTITY
+                    )
+                """
+            else:
+                create_temp_sql = f"""
+                    CREATE TABLE {temp_destination_table} (
+                        {", ".join([f'{key} {val}' for key, val in source_schema.items()])}
+                    )
+                """
+            with db.connection.cursor() as cursor:
+                logger.info(f"Creating {temp_destination_table}")
+                cursor.execute(create_temp_sql)
+                logger.info(f"{temp_destination_table} created.")
+
+                # Copy over the constraints before indexes
+                # Note: we could of included indexes above (`INCLUDING CONSTRAINTS`) but we need to drop the
+                #       foreign key ones
+                if source_table:
```

Comment:
I was able to find a scenario where `source_table` is `None`. In the `TABLE_SPEC` for `award_search` we don't specify a `source_table` to read from because we're not writing that data from Postgres to Delta Lake, but we want the ability to write it back to Postgres.
I'm not sure we're going to get the expected behavior here though because we'd like to copy constraints from `award_search` to `award_search_temp`, which won't happen based on this if statement.

Reply:
Yeah this is a kind of rub I need to square with other devs. If we want to copy indexes, constraints, etc. from SQL (not already a table in postgres), we need to update the TABLE_SPEC to include something that specifies these. I want to avoid having to regex SQL scripts to find said metadata.